### PR TITLE
Create Table of Contents in README

### PR DESCRIPTION
**Pull request by AjayJagan**
To create table of contents so as to navigate easily to specific sections of readme.

## Description
Adding a TOC section to to the README.md.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -3,6 +3,28 @@ This operator is the primary operator for Open Data Hub. It is responsible for e
 Jupyter Notebooks, Modelmesh serving, Datascience pipelines etc. The operator makes use of `DataScienceCluster` CRD to deploy
 and configure these applications.
 
+### Table of contents
+- [Usage](#usage)
+  - [Installation](#installation)
+- [Dev Preview](#dev-preview)
```
Comment (ajaypratap003) [inline README.md:9]: I think `Dev Preview` should be sibling not parent tag. There is a separate link for Dev readme file.

**Review by ajaypratap003 — COMMENTED** [review]
Added a suggestion. Changes looks good. 
One thing I noticed that direct links are not working properly when I click on link It redirect me on top of the readme instead selected tag. 
Please confirm  whether redirect links are working as expected or not.

File: README.md (hunk 1 above)
Comment (AjayJagan) [reply README.md:9]: Thanks for the review @ajaypratap003  👍 . so When I see the Dev Preview markdown it has `## Dev Preview`
Which is on the same level as Usage right? Please correct me if I am wrong

**Comment by AjayJagan** [comment]
> Added a suggestion. Changes looks good. One thing I noticed that redirect links are not working properly when I click on link It redirect me on top of the readme instead selected tag. Please confirm whether redirect links are working as expected or not.

I checked in my local VS code preview and the links seem to work fine for me

File: README.md (hunk 1 above)
Comment (ajaypratap003) [reply README.md:9]: If it's intentionally mark as parent then fine.

**Review by ajaypratap003 — APPROVED** [review]
LGTM

**Review by VaishnaviHire — APPROVED** [review]
/lgtm

**Pull request merged**